### DISCUSSION TITLE
Can I wrap an observable returned from mobx?

### DISCUSSION THREAD

**Discussion by InputNeuron** [mobxjs/mobx#2882] [General]
I have an observable I'm using in an observer.  however, I want to wrap it so that I can override some of the behavior as I'm building an undo/redo store.

Will the observer work with the observable even though it's wrapped now?

**Comment by mweststrate** [comment]
Wrapping is a pretty vague term, I'm not sure what you mean by it. But,
just try it first in a small experiment?

**Comment by InputNeuron** [comment]
Let's say I have an observable with an @computed method...

so it's has a getMessage()  and a setMessage method

I then create a wrapped version that has an internal instance variable of the original that I delegate methods to.

```typescript
so somethign like

class FooInstance {
   constructor(private readonly delegate: MainInstance);

  public setMessage(message: string) {
    this.delegate.setMessage(message);
  } 

  public getMessage() {
    return this.delegate.getMessage();
  } 

}
```

... something like that.

**Comment by mweststrate** [comment]
Yeah no problem, MobX will track through that.
